lib/associations/associations.rb

```diff
@@ -2,11 +2,35 @@ module ActiveHash
   module Associations
 
     module ActiveRecordExtensions
-
       def self.extended(base)
         require_relative 'reflection_extensions'
       end
 
+      def has_many(association_id, **options)
+        if options[:through]
+          klass_name = association_id.to_s.classify
+          klass =
+            begin
+              klass_name.constantize
```

Comment:
This makes me nervous.
I believe has_many is executed within a lock since it is defining a class.

Think this issue explains the issue: https://github.com/rails/rails/issues/34310

In our projects, when we are defining a class, we work hard to not define a second one.
Not sure if this race condition still exists.

Reply:
ok, this is the same code path as our belongs_to fix.

I wasn't able to follow the association builder code to find an answer
looks like they do call safe_constantize (sparingly)

Reply:
Hmm. I'll create a separate PR converting all our invocations to `safe_constantize`.

Reply:
🤦 I definitely did not read carefully enough. Apologies for missing your point, I get it now.